### DISCUSSION TITLE
Turbo 2.0 UI - Organize tasks in task list by workspaces and view workspace logs

### DISCUSSION THREAD

**Discussion by foxted** [vercel/turborepo#8988] [Ideas]
### Goals

1. Easier to see tasks at a glance
2. Productivity

Toggling between seeing logs for a single task vs a whole workspace would be nice too. 

### Background

Currently, when running the UI, tasks are ordered randomly.

### Proposal

It would improve the developer experience if the tasks could be grouped by workspaces instead and we'd have the ability to toggle between:

- Viewing the logs for a single task
- Viewing the logs for a whole workspace (like the existing stream view, but filtered for a given workspace)
- View all the logs together (like the existing stream view)

For example: 

```javascript 
apps // <- selecting this would show all the logs for the apps/ workspace
    - apps/app-a // <- selecting this would only show the logs for apps/app-a
    - apps/app-b // <- selecting this would only show the logs for apps/app-b
packages:  // <- selecting this would show all the logs for the packages/ workspace
    - packages/package-a  // <- selecting this would only show the logs for packages/package-a
    - packages/package-b  // <- selecting this would only show the logs for packages/package-b
 all // <- selecting this would show all the logs
```

**Comment by anthonyshew** [comment]
Grouping by directories probably isn't wise, since you don't necessarily have to use the structure in the example. You could have a wealth of top-level directories and end up with the same problem. Alternatively, you could be using top-level directories as packages in the first place. And, further, you could be using some combination of the two!

However, I do like two other parts of this:
-  Being able to select all logs: At times, this is just more convenient, for situations where you want to watch for the timing of something, for instance
- Some way of creating a deterministic order: It's possible for us to create a specific order, but the problem is, order with what? We've discussed using the task graph or going alphabetical, but both of those options come with their gaps. Would be interested to hear your thoughts.

**Reply by foxted** [reply]
Makes sense. 

I think a simple alphabetical order could already help (particularly when using a prefix based naming convention). 

For the directory grouping, I assumed we could use the workspace definition as a guide. Haven't had a project with top-level directories, so I'm not sure how the workspace config would look like for such a project.
